### PR TITLE
Bridge C++ library core to PySCF

### PR DESCRIPTION
**Pull request by pi246**
Please let me know if there is any modification I should make! Thanks!

### CODE REVIEW

File: src/pyscf_losc/utils.py [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+            return E_tot
+
+        loscmf.get_fock = get_fock
+        loscmf.energy_tot = energy_tot
+
+    return loscmf
+
+
+
+
```
Comment (zc62) [inline src/pyscf_losc/utils.py]: Extra empty lines

File: src/pyscf_losc/losc_options.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        """Print all LOSC options to PySCF output file."""
+        rst = []
+        rst.append('')
+        rst.append('==> LOSC Settings <==')
+        for module, opts in self._options.items():
+            rst.append(f'--> module={module}')
+            for key, val in opts.items():
+                rst.append(f'    {key}: {val}')
+            rst.append('')
+        return '\n'.join(rst)
```
Comment (zc62) [inline src/pyscf_losc/losc_options.py:157]: Configure your editor to always insert a newline, even if it is the end of file. This happens to all new files you created in this commit.
If using vscode, configure in this way:
```json
"files.insertFinalNewline": true,
```
Some others I use along with this setting:
```json
"files.trimTrailingWhitespace": true,
"files.trimFinalNewlines": true,
```

Or did you use any code formatter? If so, find corresponding configurations for that.

File: src/pyscf_losc/utils.py [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+            occ_idx.append(tuple())
+            occ_val.append(tuple())
+
+    nocc = tuple(len(x) for x in occ_idx)
+
+    return nocc, occ_idx, occ_val
+
+
+
+
```
Comment (zc62) [inline src/pyscf_losc/utils.py]: Extra empty lines. I think code formatters should catch these?

**Review by zc62 — CHANGES_REQUESTED** [review]

**Comment by zc62** [comment]
The code looks great, though I cannot really tell if it works as intended or not. I think it would be best if you can add unit tests. @Miocbb How did you do unit tests for the psi4 plugin?

In addition, I think proper documentation about how to use this plugin is required, otherwise people who are not familiar with this cannot easily use it. Refer to how @Miocbb wrote documentations for the psi4 plugin. See a bunch of rst files under https://github.com/Yang-Laboratory/losc/tree/2405d7b1e7d1d8f593403573f6af7a0b6b988c5e/docs_source/sphinx

**Comment by zc62** [comment]
And you can try to write more standardized git commit messages as well. [Here is what I saw and took as a good standard for commit messages.](https://github.com/mpv-player/mpv/blob/master/DOCS/contribute.md#write-good-commit-messages) You can also refer to the commit history of this repository and see how Miocbb wrote them.

**Comment by Miocbb** [comment]
> The code looks great, though I cannot really tell if it works as intended or not. I think it would be best if you can add unit tests. @Miocbb How did you do unit tests for the psi4 plugin?
> 
> In addition, I think proper documentation about how to use this plugin is required, otherwise people who are not familiar with this cannot easily use it. Refer to how @Miocbb wrote documentations for the psi4 plugin. See a bunch of rst files under https://github.com/Yang-Laboratory/losc/tree/2405d7b1e7d1d8f593403573f6af7a0b6b988c5e/docs_source/sphinx

I made some simple tests for psi4_losc under losc/test/psi4_losc. @pi246

**Comment by zc62** [comment]
Not sure why this PR got merged without further review, as apparently 65099b36f97a9dfb7f7aa9b33ecf2407850ca9a5 got back those format issues that e42faebb5608290262f9e375ed1e7cb842cadfaf tried to resolve, for example. @pi246

**Pull request merged**